Framework/PythonInterface/plugins/algorithms/LoadEXED.py

```diff
@@ -0,0 +1,189 @@
+                                          action=FileAction.Load,
+                                          extensions = ["raw"]),
+                             doc="Data file produced by egraph.")
+
+        self.declareProperty(FileProperty(name="InstrumentXML",defaultValue = "",
+                                          action=FileAction.OptionalLoad,
+                                          extensions = ["xml"]),
+                             doc="Instrument definition file. If no file is specified, the default idf is used.")
+
+        self.declareProperty(MatrixWorkspaceProperty(name="OutputWorkspace",
+                                                     defaultValue = "",
+                                                     direction=Direction.Output),
+                             doc="Mantid workspace containing the measured data.")
+
+    def PyExec(self):
+        from mantid.simpleapi import LoadInstrument
+        #from mantid.simpleapi import LoadExedHelper
+        #from mantid.simpleapi import CreateWorkspace
+        from mantid.simpleapi import AddSampleLog
+        from mantid.simpleapi import SetGoniometer
+        from mantid.simpleapi import ExtractSpectra
+        from mantid.simpleapi import MaskDetectors
+        from mantid.simpleapi import RemoveMaskedSpectra
+        #from mantid.simpleapi import mtd
+
+        #self.LoadRaw()
+
+        fn = self.getPropertyValue("Filename")
+        wsn = self.getPropertyValue("OutputWorkspace")
+        #print (fn, wsn)
+
+        self.fxml = self.getPropertyValue("InstrumentXML")
+
+        #load data
+
+        parms_dict, det_udet, det_count, det_tbc, data = read_file(fn)
+        nrows=int(parms_dict['NDET'])
+        #nbins=int(parms_dict['NTC'])
+        xdata = np.array(det_tbc)
+        xdata_mon = np.linspace(xdata[0],xdata[-1], len(xdata))
+        ydata=data.astype(np.float)
+        ydata=ydata.reshape(nrows,-1)
+        edata=np.sqrt(ydata)
+        #CreateWorkspace(OutputWorkspace=wsn,DataX=xdata,DataY=ydata,DataE=edata,
+        #                NSpec=nrows,UnitX='TOF',WorkspaceTitle='Data',YUnitLabel='Counts')
+        nr,nc=ydata.shape
+        ws = WorkspaceFactory.create("Workspace2D", NVectors=nr,
+                                     XLength=nc+1, YLength=nc)
+        for i in range(nrows):
+            ws.setX(i, xdata)
+            ws.setY(i, ydata[i])
+            ws.setE(i, edata[i])
+        ws.getAxis(0).setUnit('tof')
+        AnalysisDataService.addOrReplace(wsn,ws)
+
+        #self.setProperty("OutputWorkspace", wsn)
+        #print ("ws:", wsn)
+        #ws=mtd[wsn]
+
+        # fix the x values for the monitor
+        for i in range(nrows-2,nrows):
+            ws.setX(i,xdata_mon)
+        print ("set detector IDs")
+        #set detetector IDs
+        for i in range(nrows):
+            ws.getSpectrum(i).setDetectorID(det_udet[i])
+
+        #load idf
+
+        if (self.fxml == ""):
```

Comment:
Don't need the if statement, just pass both `InstrumentName` and `Filename`. If the filename is specified, it will load that